### PR TITLE
Implement User-Specific Saved Internships Feature

### PR DESCRIPTION
**Pull request by Ememobong28**
**Description:**
This pull request introduces a new feature that allows users to save internships they are interested in for later review. This has been implemented via a new SavedInternship model, with corresponding routes and methods for creating, retrieving, and deleting saved internships.

**Features:**
1.) Users can save internships by clicking the bookmark button.
2.) Saved internships can be viewed in a separate 'Saved Internships' page.
3.) Users can remove internships from their saved internships list by un-bookmarking it. 
4.) Internships are uniquely associated with the users that save them, ensuring each user sees only their saved internships.
5.) Styling has been updated to match the existing style of the internships cards.

This contributes to my milestones of improving user experience by giving them greater control and customization over their internship search process and is a step towards my Notifications Technical Challenge.

**Milestones:**
1.) User-specific customization: This feature enables user-specific customization by letting users save internships relevant to them. This will improve the user experience and encourage repeat visits to TechHive.

2.) Improved functionality: The ability to save internships for later review adds another layer of functionality to our site. This aligns with our objective of providing comprehensive internship search features.

3.) Increased engagement: The 'Saved Internships' feature increases user engagement by providing an additional interaction point.

**Test Plan:**
1.) Save Internship: Check that the bookmark save button correctly saves an internship. The internship should immediately appear in the user's saved internships page.
2.) Delete Internship: Check that the un-bookmark button correctly removes a saved internship from the saved internships list.
3.) Page Display: The saved internships page should accurately display all of a user's saved internships, including all relevant details.
4.) Persistence: After saving internships, restart the server and confirm that saved internships persist and are still visible in the user's saved internships list.
5.) Edge Case Testing: Test the feature with no internships saved, with multiple internships saved, with the same internship saved multiple times, and with internships that have been removed.

**Video-Recordings**
1.) Saved Internships Video Recording in Workplace Group Chat
2.) Postman check saved internships

Screenshot 2023-07-20 at 2 27 19 PM

### CODE REVIEW

File: techhive-ui/src/components/Internships/Internships.jsx [hunk 1]
```diff
@@ -38,7 +39,8 @@ function Internships() {
         const response = await fetch('http://localhost:3000/internships');
         const data = await response.json();
         data.forEach(internship => {
-          internship.isLiked = false; 
+          internship.isLiked = false;
+          internship.isSaved = false;
```
Comment (nicolec2626) [inline techhive-ui/src/components/Internships/Internships.jsx:43]: what's the difference between liked and saved? Will they both be used for notifications in the future? 
Will students be able to see both "liked internships" and "saved internships"?

File: techhive-ui/src/components/SavedInternships/SavedInternships.jsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                  />
+                  <div className="internship-details">
+                    <h3>{internship.title}</h3>
+                    <p>{internship.description}</p>
+                    <a href={internship.link} target="_blank" rel="noopener noreferrer">
+                      Learn More
+                    </a>
+                    <p className="company">Company: {internship.company}</p>
+                    <p className="category">Category: {internship.category.join(', ')}</p>
+                  </div>
```
Comment (nicolec2626) [inline techhive-ui/src/components/SavedInternships/SavedInternships.jsx:63]: nice clean and clear code!

File: techhive/models/savedInternships.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  },
+  internshipId: {
+    type: DataTypes.INTEGER,
+    references: {
+      model: Internship, 
+      key: 'id', 
+    }
+  },
+}, {
+  tableName: 'savedinternships'
```
Comment (nicolec2626) [inline techhive/models/savedInternships.js:23]: where are the savedinternships, users, and internships tables saved?

**Comment by nicolec2626** [comment]
This PR is also a little big - we can split it up by features, something like: 
PR 1. Users can save and unsave internships 
PR 2. Users can view saved internships in a separate 'saved internships' page
PR 3. Update styling 

Something to keep in mind for future PRs!
